### PR TITLE
Shallow water/updating solver analysis

### PR DESCRIPTION
**Pull request by miguelmaso**
the python scripts in the application are derived from the core python_solver and analysis_stage

### CODE REVIEW

File: applications/ShallowWaterApplication/python_scripts/pfem2_primitive_var_solver.py
Comment (loumalouomega) [inline applications/ShallowWaterApplication/python_scripts/pfem2_primitive_var_solver.py]: Better use the logger

File: applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py
Comment (philbucher) [inline applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py]: I would remove this method completely, it is already in the baseclass including the warning

File: applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py
Comment (philbucher) [inline applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py:208]: isnt this computed & set by CloneTimeStep?

Comment (loumalouomega) [reply applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py:208]: Yep

Comment (miguelmaso) [reply applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py:208]: I think CloneTimeStep is setting the new time

https://github.com/KratosMultiphysics/Kratos/blob/b8b4063c7d297747dd35c774b9d727650bfdd1cf/kratos/sources/model_part.cpp#L165

I am just updating the delta time variable

File: applications/ShallowWaterApplication/python_scripts/eulerian_primitive_var_solver.py
Comment (philbucher) [inline applications/ShallowWaterApplication/python_scripts/eulerian_primitive_var_solver.py]: This is "double defined", if you ask for the printing rank, them you can use the logger directly

File: applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py
Comment (philbucher) [inline applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py:41]: Note:
If you do this here, i.e. before reading the ModelPart, then the SubmodelParts will only be accessible by "MainModelPartName.SubModelpartName" etc. whenever they are requested in the Model
This happens e.g. when the Processes are constructed.

This is the way to go in the future, but so far the names of the SubModelparts are usually not having also the MainModelPartName

I am not saying you shouldn't do this, but be aware it might break examples

=> In Fluid & Structure the ModelPart is added to the Model AFTER reading it, i.e. in `PrepareModelPart` to have a temp solution to not break existing examples

**Review by philbucher — COMMENTED** [review]
two more comments

**Comment by loumalouomega** [comment]
@miguelmaso do you need approval or not yet?

File: applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py
Comment (miguelmaso) [reply applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py:41]: Thank you for the two comments. I was forgetting to update an example.
I prefer to access the SubmodelParts by the full name ("main.sub"), like the way to go in the future.

**Comment by miguelmaso** [comment]
I think it can be merged after passing the checks

**Review by loumalouomega — APPROVED** [review]
OK

**Pull request merged**
